### PR TITLE
file-upload.js modified : Remaining upload time corrected

### PR DESCRIPTION
**Pull request by sarthikadhawan**
Hi
I have tried to resolve the issue 'Remaining upload time is wrong :#3647'  . Changes are from line 976 to 1040 .

### CODE REVIEW

File: apps/files/js/file-upload.js [hunk 1]
```diff
@@ -991,7 +1005,7 @@ OC.Uploader.prototype = _.extend({
 							+ '</span><span class="mobile">'
 							+ t('files', '...')
 							+ '</span></em>');
-					$('#uploadprogressbar').tooltip({placement: 'bottom'});
+                    $('#uploadprogressbar').tipsy({gravity:'n', fade:true, live:true});
```
Comment (ChristophWurst) [inline apps/files/js/file-upload.js]: please use tabs instead of spaces

File: apps/files/js/file-upload.js [hunk 2]
```diff
@@ -1014,11 +1028,49 @@ OC.Uploader.prototype = _.extend({
...
-					$('#uploadprogressbar .label .mobile').text(h);
-					$('#uploadprogressbar .label .desktop').text(h);
+					var smoothRemainingSeconds;
+					if(bufferIndex2<20)
+						smoothRemainingSeconds = (bufferTotal/bufferIndex2);
+					else
+						 smoothRemainingSeconds = (bufferTotal / bufferSize); 
+					var date = new Date(smoothRemainingSeconds * 1000);
+					var timeStringDesktop = "";
+					var timeStringMobile = "";
```
Comment (ChristophWurst) [inline apps/files/js/file-upload.js]: use single quotes

File: apps/files/js/file-upload.js [hunk 3]
```diff
@@ -1014,11 +1028,49 @@ OC.Uploader.prototype = _.extend({
...
+						bufferIndex2 = (bufferIndex2 + 1);
+
+
 					}
-					var smoothRemainingSeconds = (bufferTotal / bufferSize); //seconds
-					var h = moment.duration(smoothRemainingSeconds, "seconds").humanize();
-					$('#uploadprogressbar .label .mobile').text(h);
-					$('#uploadprogressbar .label .desktop').text(h);
+					var smoothRemainingSeconds;
+					if(bufferIndex2<20)
```
Comment (ChristophWurst) [inline apps/files/js/file-upload.js]: use curly braces for the if and else branch

**Review by ChristophWurst — COMMENTED** [review]
some nitpicks :)

File: apps/files/js/file-upload.js (hunk 1 above)
Comment (jancborchardt) [inline apps/files/js/file-upload.js]: Also, didn’t we change from tipsy to tooltip?

Comment (ChristophWurst) [inline apps/files/js/file-upload.js]: Yup! @xx621998xx please revert this change

**Comment by MorrisJobke** [comment]
And remove the executable bit from the JS file (10644 -> 10755 in the git diff indicates, that the executable flag was set)

File: apps/files/js/file-upload.js (hunk 3 above)
Comment (MorrisJobke) [inline apps/files/js/file-upload.js]: ... and spaces around the operator ;)

**Comment by MorrisJobke** [comment]
Beside that it looks good :)

File: apps/files/js/file-upload.js [hunk 4]
```diff
@@ -1010,12 +1011,23 @@ OC.Uploader.prototype = _.extend({
 					lastSize = data.loaded;
 					diffSize = diffSize / diffUpdate; // apply timing factor, eg. 1mb/2s = 0.5mb/s
 					var remainingSeconds = ((data.total - data.loaded) / diffSize);
-					if(remainingSeconds >= 0) {
+					if(remainingSeconds >= 0) 
+					{
 						bufferTotal = bufferTotal - (buffer[bufferIndex]) + remainingSeconds;
 						buffer[bufferIndex] = remainingSeconds; //buffer to make it smoother
 						bufferIndex = (bufferIndex + 1) % bufferSize;
+						bufferIndex2 = (bufferIndex2 + 1);
```
Comment (nickvergessen) [inline apps/files/js/file-upload.js]: Just do `bufferIndex2++;` instead?

File: apps/files/js/file-upload.js [hunk 5]
```diff
@@ -1010,12 +1011,23 @@ OC.Uploader.prototype = _.extend({
...
 					}
-					var smoothRemainingSeconds = (bufferTotal / bufferSize); //seconds
+					var smoothRemainingSeconds;
+					if(bufferIndex2<20)
+					{
+							smoothRemainingSeconds = (bufferTotal/bufferIndex2);
```
Comment (nickvergessen) [inline apps/files/js/file-upload.js]: please remove the brackets

File: apps/files/js/file-upload.js [hunk 6]
```diff
@@ -1010,12 +1011,23 @@ OC.Uploader.prototype = _.extend({
...
+					}
+					else
+					{
+						 smoothRemainingSeconds = (bufferTotal / bufferSize);
```
Comment (nickvergessen) [inline apps/files/js/file-upload.js]: same here

**Review by nickvergessen — CHANGES_REQUESTED** [review]
Executable bit needs to go.

File: apps/files/js/file-upload.js [hunk 7]
```diff
@@ -1010,12 +1011,23 @@ OC.Uploader.prototype = _.extend({
 					lastSize = data.loaded;
 					diffSize = diffSize / diffUpdate; // apply timing factor, eg. 1mb/2s = 0.5mb/s
 					var remainingSeconds = ((data.total - data.loaded) / diffSize);
-					if(remainingSeconds >= 0) {
+					if(remainingSeconds >= 0) 
+					{
```
Comment (MorrisJobke) [inline apps/files/js/file-upload.js]: brackets in one line:
```js
if(remainingSeconds >= 0) {
```

File: apps/files/js/file-upload.js [hunk 8]
```diff
@@ -1010,12 +1011,23 @@ OC.Uploader.prototype = _.extend({
...
 						bufferTotal = bufferTotal - (buffer[bufferIndex]) + remainingSeconds;
 						buffer[bufferIndex] = remainingSeconds; //buffer to make it smoother
 						bufferIndex = (bufferIndex + 1) % bufferSize;
+						bufferIndex2 = bufferIndex2++;
 					}
-					var smoothRemainingSeconds = (bufferTotal / bufferSize); //seconds
+					var smoothRemainingSeconds;
+					if(bufferIndex2<20)
```
Comment (MorrisJobke) [inline apps/files/js/file-upload.js]: brackets in one line, space after if, spaces around operator:
```js
if (bufferIndex2 < 20) {
```

File: apps/files/js/file-upload.js [hunk 9]
```diff
@@ -1010,12 +1011,23 @@ OC.Uploader.prototype = _.extend({
...
+					{
+							smoothRemainingSeconds = bufferTotal / bufferIndex2;
```
Comment (MorrisJobke) [inline apps/files/js/file-upload.js]: Indentation is off - seems to be spaces instead of tabs.

File: apps/files/js/file-upload.js [hunk 10]
```diff
@@ -1010,12 +1011,23 @@ OC.Uploader.prototype = _.extend({
...
+					}
+					else
```
Comment (MorrisJobke) [inline apps/files/js/file-upload.js]: Brackets on same line:

```js
} else {
```

**Comment by MorrisJobke** [comment]
Sorry for nitpicking.

**Comment by nickvergessen** [comment]
I removed the executable bit and added some more hardening in 
https://github.com/nextcloud/server/pull/3896

Thanks for your contribution @xx621998xx  🎉

**Pull request closed without merging**